src/routing_node.rs

```diff
@@ -78,6 +80,7 @@ pub struct RoutingNode {
     public_id_cache     : LruCache<NameType, PublicId>,
     connection_cache    : BTreeMap<NameType, SteadyTime>,
     accumulator         : MessageAccumulator,
+    data_cache          : LruCache<NameType, Vec<u8>>,
```

Comment:
we can store the actual types in the cache, without a need to serialise them into Vec<u8>